GetDKAN/dkan-tools
# Clean up docker-compose YAML and fix dkan:make

A number of improvements coming out of the last few weeks of work w/USDA. 

Main fix: Removes unnecesary docker compose YAML files, simplifies the dktl script and allows us to use overrides against docker-compose rather than replace the whole common file.

To override the common file, create src/docker/docker-compose.overrides.yml and add any overrides there.
Also: 

* Fixes an issue with the dkan make overrides, which were not applying
* Renames dktl.sh to just dktl. This makes it easier to simply add the dkan-tools/bin dir to the `$PATH` rather than having to create a symbolic link.
* Adds a check for TTY and adds a `-T` flag to `docker-compose exec` commands if appropriate (not included in QA steps; could not think of how to QA for this, but see https://github.com/GetDKAN/dkan-tools/pull/68/commits/7dde05ec1a432cc9e1991c83b273f250b9d9af4b).
* Updates robo to 1.3.2, which allows us to share environment variables with the exec tasks (which in turn allows us to set `AWS_ACCESS_KEY_ID` and `AWS_SECRET_ACCESS_KEY` environment variables rather than having to create a ~/.aws/credentials file)

**NOTE:** The changing of `dktl.sh` is obviously a breaking change. Be sure to update your `$PATH`/symlinks.

## QA Steps

1. Create a new DKTL site for DKAN
2. Create src/docker/docker-compose.overrides.yml using the yaml code below
3. Bring up the docker containers and confirm that the web container is using PHP 5.6 instead of PHP 7.1.
4. Add the make file below as src/make/dkan.make
5. Run `dktl dkan:make` and confirm that dkan_paragraphs is added to dkan/modules/contrib
6. To test the environment variables, destroy your docker containers and remove your ~/.aws dir.
7. Run `export AWS_ACCESS_KEY_ID=[your access id]` and `export AWS_SECRET_ACCESS_KEY=[your secret]`
8. Set up .dktl with a valid aws `db_url` 

## Docker compose file:

```yaml
version: "2"
services:
  web:
    image: nuams/drupal-apache-php:1.0-5.6
```

## Makefile
```yaml
api: 2
core: 7.x
projects:
  dkan_dataset_paragraphs:
    download:
      type: git
      url: https://github.com/GetDKAN/dkan_dataset_paragraphs.git
      branch: 7.x-1.x
  ```
